Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EpisodeDataManager.swift

```diff
@@ -728,16 +728,15 @@ class EpisodeDataManager {
     }
 
     func markAllSynced(episodes: [Episode], dbQueue: FMDatabaseQueue) {
-        if episodes.count == 0 { return }
+        guard !episodes.isEmpty else {
+            return
+        }
 
         dbQueue.inDatabase { db in
             do {
                 db.beginTransaction()
-
-                for episode in episodes {
-                    try db.executeUpdate("UPDATE \(DataManager.episodeTableName) SET playingStatusModified = 0, playedUpToModified = 0, durationModified = 0, keepEpisodeModified = 0, archivedModified = 0 WHERE id = ?", values: [episode.id])
-                }
-
+                let ids = episodes.map({"\($0.id)"})
+                try db.executeUpdate("UPDATE \(DataManager.episodeTableName) SET playingStatusModified = 0, playedUpToModified = 0, durationModified = 0, keepEpisodeModified = 0, archivedModified = 0 WHERE id IN (\(ids.joined(separator: ",")))", values: nil)
```

Comment:
Should we feature flag this change? I know we've had performance issues with filters when doing queries with long lists so I don't know if that might be an issue here. Maybe less likely because those were issues on the main thread.

Reply:
+1